xcdat/bounds.py

```diff
@@ -138,98 +139,155 @@ def add_missing_bounds(self, width: float = 0.5) -> xr.Dataset:
         -------
         xr.Dataset
         """
-        axes = CF_NAME_MAP.keys()
+        axes = CF_ATTR_MAP.keys()
 
         for axis in axes:
-            # Check if the axis coordinates can be mapped to.
+            # Get all of the coordinate variables that can be mapped to.
             try:
-                get_axis_coord(self._dataset, axis)
+                coords = get_axis_coord(self._dataset, axis)
             except KeyError:
                 continue
 
-            # Determine if the axis is also a dimension by determining if there
-            # is overlap between the CF axis names and the dimension names. If
-            # not, skip over axis for validation.
-            if len(set(CF_NAME_MAP[axis]) & set(self._dataset.dims.keys())) == 0:
-                continue
-
-            # Check if bounds also exist using the "bounds" attribute.
-            # Otherwise, try to add bounds if it meets the function's criteria.
-            try:
-                self.get_bounds(axis)
-                continue
-            except KeyError:
-                pass
-
-            try:
-                self._dataset = self.add_bounds(axis, width)
-            except ValueError:
-                continue
+            # Loop over the coordinate variables and attempt to get bounds.
+            # If bounds aren't found, attempt to add bounds (must not be a
+            # singleton).
+            for coord in coords.coords.values():
+                try:
+                    self.get_bounds(axis, str(coord.name))
+                    continue
+                except KeyError:
+                    pass
+
+                try:
+                    self._dataset = self.add_bounds(axis, width)
```

Comment:
Should we be making a clone of the dataset?

Reply:
No, `self._dataset` needs to be updated every loop because `self.add_bounds()` makes a copy of `self._dataset()` and returns it. If we don't update `self._dataset` directly, bounds for the previous iterations won't be saved.